### DISCUSSION TITLE
Why does the RAC Select component render items on mount (without being opened)?

### DISCUSSION THREAD

**Discussion by marbemac** [adobe/react-spectrum#5824] [General]
Seems to cause a decent amount of blocking if there are many selects on the screen. For example, imagine a members table with 100 organization members, where each row renders some selects to do things like change the member role, etc.

**Comment by snowystinger** [comment]
That's probably the hidden select that backs the component and allows for features such as autofill. I don't think we'd want to prevent that. Eventually, we want to support virtualization in RAC, which should alleviate some of this issue, as having 100 rows rendered at once would be uncommon.

**Reply by marbemac** [reply]
Doesn't seem to be the HiddenSelect - seems to be this -> https://github.com/adobe/react-spectrum/blob/main/packages/react-aria-components/src/Select.tsx#L130-L141

Is it that uncommon? Mind you, I'm not talking 100 rows in the select, but rather a dense UI that itself renders many selects. For example, the Figma organization members dashboard (or many SaaS dashboards that involve lists/grids of members / projects etc and associated selects, checkboxes, and menus to act on each row).

**Reply by snowystinger** [reply]
Yeah, the reason for that render is that it creates the collection, allowing us to populate the hidden select and do typeahead and autofill etc.

As for common/uncommon, I don't know of other libraries that do this, but it's what drives our Collection API. We could look into speeding up that initial collection building render, but we'll need to track down what about it is truly slow. It's just using react in a fake minimal dom so that we get quick diffing and fast updates. The initial render shouldn't be all that slow though because there is no paint step or layout to compute, but maybe something unexpected is happening. It'll need more investigation.

**Reply by marbemac** [reply]
>Yeah, the reason for that render is that it creates the collection, allowing us to populate the hidden select and do typeahead and autofill etc.

Hmm got it - is typeahead/autofill etc needed before the select is focused and/or opened?

Thanks for the reply btw!

**Reply by snowystinger** [reply]
Yep, auto-fill can happen at any time. Typeahead, not so much.

**Reply by marbemac** [reply]
Got it - in that case there's probably not much to be done. Thank you!

**Reply by devongovett** [reply]
Do you have a profile showing a real performance issue here or is this more theoretical?
